### ISSUE TITLE
submit_form giving both form_number   and with_fields

### ISSUE THREAD

**Issue by epa** [libwww-perl/WWW-Mechanize#32]
`submit_form` takes various filters such as `form_number` and `with_fields`.  The  synopsis in the pod documentation gives an example usage:
````
            $mech->submit_form(
                form_number => 3,
                fields      => {
                    username    => 'mungo',
                    password    => 'lost-and-alone',
                }
            );
````
(side buglet: this has `fields` instead of `with_fields`)

However, if `with_fields` is passed then any `form_number` argument is ignored; similarly if you pass `form_number` then `form_name` is ignored, and so on.

The behaviour that makes the most sense is to AND together the filters  provided.  In a way, that would be a backwards-incompatible change since  there may be code in the wild that passes both `form_number` and `with_fields`  and has come to rely on the fact that `form_number` is ignored in this case.  So you might want to issue a warning (during a transitional period) if  multiple filters are passed.

(Previously filed at https://code.google.com/p/www-mechanize/issues/detail?id=264 but that page is now broken)

**Comment by oalders** [comment]
I would accept a patch for this.

**Comment by epa** [comment]
https://github.com/libwww-perl/WWW-Mechanize/pull/50

**Comment by epa** [comment]
Fixed in 1.80, thanks.
